deploy/cloudformation/manifest-pipeline-pipeline.yml

```diff
@@ -0,0 +1,769 @@
+      RoleArn: !GetAtt
+        - CodePipelineTrustRole
+        - Arn
+      Stages:
+      - Name: Source
+        Actions:
+        - Name: SourceFromRepo
+          InputArtifacts: []
+          ActionTypeId:
+            Version: "1"
+            Category: Source
+            Owner: ThirdParty
+            Provider: GitHub
+          OutputArtifacts:
+          - Name: SourceFromRepo
+          Configuration:
+            Repo: !Ref GitHubRepoName
+            Branch: !Ref GitHubRepoBranch
+            OAuthToken: !Ref GitHubToken
+            Owner: !Ref GitHubUser
+          RunOrder: 1
+        -
+          Name: "RetrieveConfigCode"
+          ActionTypeId:
+            Owner: ThirdParty
+            Category: Source
+            Provider: GitHub
+            Version: "1"
+          Configuration:
+            Repo: !Ref ConfigurationRepoName
+            Branch: !Ref ConfigurationRepoBranchName
+            Owner: !Ref GitHubUser
+            OAuthToken: !Ref GitHubToken
+            PollForSourceChanges: false
+          OutputArtifacts:
+            - Name: ConfigCode
+          RunOrder: 1
+      - Name: Build
+        Actions:
+        - Name: BuildFromRepo
+          InputArtifacts:
+          - Name: SourceFromRepo
+          - Name: ConfigCode
+          ActionTypeId:
+            Category: Build
+            Owner: AWS
+            Version: "1"
+            Provider: CodeBuild
+          OutputArtifacts:
+          - Name: BuildFromRepo
+          Configuration:
+            ProjectName: !Sub "${AppName}-build"
+            PrimarySource: SourceFromRepo
+          RunOrder: 1
+
+      - Name: DeployToTest
+        Actions:
+          - Name: CreateChangeSet
+            ActionTypeId:
+              Category: Deploy
+              Owner: AWS
+              Provider: CloudFormation
+              Version: '1'
+            InputArtifacts:
+              - Name: BuildFromRepo
+            Configuration:
+              ActionMode: CHANGE_SET_REPLACE
+              Capabilities: CAPABILITY_IAM
+              RoleArn: !GetAtt
+                - CloudFormationTrustRole
+                - Arn
+              StackName: !Sub "${AppName}-test"
+              ChangeSetName: TestChangeSetName
+              TemplateConfiguration: !Sub "BuildFromRepo::${TestStackConfig}"
+              TemplatePath: !Sub "BuildFromRepo::${SAMOutputFile}"
+            RunOrder: '1'
+          - Name: ExecuteChangeSet
+            ActionTypeId:
+              Category: Deploy
+              Owner: AWS
+              Provider: CloudFormation
+              Version: '1'
+            Configuration:
+              ActionMode: CHANGE_SET_EXECUTE
+              Capabilities: CAPABILITY_IAM
+              ChangeSetName: TestChangeSetName
+              RoleArn: !GetAtt
+                - CloudFormationTrustRole
+                - Arn
+              StackName: !Sub "${AppName}-test"
+            RunOrder: '2'
+
+      - Name: Approval
```

Comment:
The `Approval` at least should be part of the `DeployToTest` stage, otherwise if a concurrent execution happens, the approval or rejection may not match the actual test env.